src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -93,21 +74,27 @@ BoundExpression tryOptimizeOperand(BoundExpression operand)
             }
         }
 
-        private BoundExpression LiftRangeExpression(BoundRangeExpression node, ImmutableArray<BoundExpression> operands)
+        private BoundExpression LiftRangeExpression(BoundRangeExpression node, BoundExpression left, BoundExpression right)
         {
             Debug.Assert(node.Type.IsNullableType());
-            Debug.Assert(operands.Any(operand => operand.Type.IsNullableType()));
-            Debug.Assert(operands.Length == 1 || operands.Length == 2);
+            Debug.Assert(left?.Type.IsNullableType() == true || right?.Type.IsNullableType() == true);
+            Debug.Assert(!(left is null && right is null));
 
-            ArrayBuilder<BoundExpression> sideeffects = ArrayBuilder<BoundExpression>.GetInstance();
-            ArrayBuilder<LocalSymbol> locals = ArrayBuilder<LocalSymbol>.GetInstance();
-            ArrayBuilder<BoundExpression> arguments = ArrayBuilder<BoundExpression>.GetInstance();
+            var sideeffects = ArrayBuilder<BoundExpression>.GetInstance();
+            var locals = ArrayBuilder<LocalSymbol>.GetInstance();
 
-            // left.HasValue && right.HasValue
+            // if (left.HasValue && right.HasValue)
+            //    makeRange(left.GetValueOrDefault(), right.GetValueOrDefault())
```

Comment:
>makeRange(left.GetValueOrDefault(), right.GetValueOrDefault()) [](start = 18, length = 62)

This comment doesn't match other comments below:
`// left.HasValue && right.HasValue ? new Nullable(method(left.GetValueOrDefault(), right.GetValueOrDefault())) : default`